### PR TITLE
feature: Enable folder upload

### PR DESCRIPTION
**Pull request by fregataa**
Enable vfolder upload function to allow you to upload folders by making parent directory of a target path.

### CODE REVIEW

File: src/ai/backend/storage/api/client.py [hunk 1]
```diff
@@ -277,7 +277,8 @@ async def tus_upload_part(request: web.Request) -> web.Response:
                     parent_dir = vfpath / dst_dir
                 if not parent_dir.exists():
                     parent_dir.mkdir(parents=True, exist_ok=True)
-                target_path = parent_dir / token_data["relpath"]
+                target_path: Path = parent_dir / token_data["relpath"]
+                target_path.parent.mkdir(parents=True, exist_ok=True)
```
Comment (agatha197) [inline src/ai/backend/storage/api/client.py]: I think 278~279 lines can be removed because the 281 line can cover it.

**Review by Sujin-Kim1 — CHANGES_REQUESTED** [review]
I commented minor one. But it works!

File: src/ai/backend/storage/api/client.py (hunk 1 above)
Comment (fregataa) [reply src/ai/backend/storage/api/client.py]: thanks! I updated it.

**Review by Sujin-Kim1 — APPROVED** [review]
LGTM

**Review by lizable — CHANGES_REQUESTED** [review]
folder upload fails when the directory is empty. @fregataa Could you check it and fix it?
Here's the directory tree that I tested.
Screenshot 2023-04-19 at 6 01 31 PM

**Review by achimnol — APPROVED** [review]

**Pull request merged**